src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,27 @@
 package core.basesyntax;
 
+import core.basesyntax.io.CsvReader;
+import core.basesyntax.io.CsvWriter;
+import core.basesyntax.model.Operation;
+import core.basesyntax.model.OperationType;
+import core.basesyntax.model.Report;
+
+import java.util.List;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private final CsvReader reader = new CsvReader();
+    private final CsvWriter writer = new CsvWriter();
 
+    public void getStatistic(String fromFileName, String toFileName) {
+        List<Operation> operations = reader.readOperations(fromFileName);
+        Report report = new Report();
+        for (Operation operation : operations) {
+            if (operation.getOperationType().equals(OperationType.SUPPLY)) {
+                report.incrementSupply(operation.getAmount());
+            } else {
+                report.incrementBuy(operation.getAmount());
+            }
+        }
+        writer.writeReportToCsv(report, toFileName);
     }
```

Comment:
The method 'getStatistic' seems to be doing too much by reading operations, processing them, and writing the report. Consider dividing the logic into smaller, more focused methods, for example, a method to process operations into a report and another method to handle file writing.